tasks/link-checker.js

```diff
@@ -62,11 +103,18 @@ const retrieveLinks = async (siteMapUrls, visitedLinks, localDomain) => {
       let response = await page.goto(url, { waitUntil: 'domcontentloaded' });
       await new Promise((r) => setTimeout(r, 100)); // localhost hangs on wait for idle so use a short timeout instead
       if (response && response.status() && response.status() === 200) {
-        visitedLinks[url] = true;
-
+        // Get all the links from the page
         const urlList = await page.evaluate(async (url) => {
-          let urls = [];
-          let elements = document.getElementsByTagName('a');
+          const urls = [];
+          let elements;
+          if (url === 'https://docs.amplify.aws/') {
+            // On the homepage, grab all the links (nav bar, footer, left menu)
+            elements = document.getElementsByTagName('a');
+          } else {
+            // On any other page, grab only the links in the main element
+            elements = document.querySelector('main').getElementsByTagName('a');
+          }
```

Comment:
I didn't want to keep grabbing the links from the left menu, footer, and top nav each page we visited so I split it up into these selectors. Are there pages where we should be grabbing all the `a` tags?